src/EditorFeatures/CSharpTest/Diagnostics/GenerateConstructor/GenerateConstructorTests.cs

```diff
@@ -872,5 +872,45 @@ public bool M(string s, int i, bool b)
 parseOptions: TestOptions.Regular.WithTuplesFeature(),
 withScriptOption: true);
         }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsGenerateConstructor)]
+        [WorkItem(12147, "https://github.com/dotnet/roslyn/issues/12147")]
+        public async Task TestOutVariableDeclaration_ImplicitlyTyped()
+        {
+            await TestAsync(
+@"class C { void M() { new [|C|](out var a); } }",
+@"class C { public C(out object a) { a = null; } void M() { new C(out var a); } }",
+                                parseOptions: TestOptions.Regular.WithOutVarFeature());
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsGenerateConstructor)]
+        [WorkItem(12147, "https://github.com/dotnet/roslyn/issues/12147")]
+        public async Task TestOutVariableDeclaration_ImplicitlyTyped_NamedArgument()
+        {
+            await TestAsync(
+@"class C { void M() { new C([|b|]: out var a); } }",
+@"class C { public C(out object b) { b = null; } void M() { new C(b: out var a); } }",
+                                parseOptions: TestOptions.Regular.WithOutVarFeature());
```

Comment:
Actually, a couple of scenarios are affected by the parse options because some semantic errors are suppressed due to the presence of parse errors. I will open an issue track this separately.   

---
In reply to: [68117890](https://github.com/dotnet/roslyn/pull/12159#discussion_r68117890) [](ancestors = 68117890,68112365)